kernel/src/main/java/org/cristalise/kernel/lifecycle/instance/Activity.java

```diff
@@ -230,7 +230,7 @@ else if (transition.getOutcome().isRequired())
                 String viewpoint = resolveViewpointName(newOutcome);
 
                 int eventID = hist.addEvent(agent, delegate, usedRole, getName(), getPath(), getType(),
-                        schema, getStateMachine(), transitionID, viewpoint).getID();
+                        schema, getStateMachine(), transitionID, viewpoint, attachment).getID();
```

Comment:
please use boolean here, i.e. `(attachment.length > 0)`

Reply:
and handle null attachment

Reply:
as far as I remember corba will initialise an empty byte array, but it is better to be on the safe side